### PR TITLE
SolverRewards 8: Fix executed amount for fee calculation

### PR DESCRIPTION
**Pull request by sunce86**
Fixes https://github.com/cowprotocol/services/pull/1279#discussion_r1118787688

Core change: 

```
let executed_amount = match order.data.kind {
   OrderKind::Buy => order.metadata.executed_buy_amount,
   OrderKind::Sell => order.metadata.executed_sell_amount_before_fees,
},
```

### CODE REVIEW

File: crates/autopilot/src/decoded_settlement.rs [hunk 1]
```diff
@@ -112,23 +112,30 @@ pub struct Order {
     pub buy_token: H160,
     pub sell_amount: U256,
     pub buy_amount: U256,
+    pub executed_amount: U256,
```
Comment (nlordell) [inline crates/autopilot/src/decoded_settlement.rs:115]: We also have an `executed_fee_amount` part of the order which would avoid the fee computation 👇.

**Comment by nlordell** [comment]
One advantage of using the `Trade` for computing the executed amounts, is that it is independent of the "loading order from database" code.

In theory, you could use the flag "is partially fillable" bit from the `flags` to check whether or not it is the trade's executed amount should be used or the full buy/sell amount.

File: crates/autopilot/src/decoded_settlement.rs (hunk 1 above)
Comment (sunce86) [reply crates/autopilot/src/decoded_settlement.rs:115]: I don't get the same results as in solver competition endpoint when I use `executed_fee_amount`. I will double check.

Comment (sunce86) [reply crates/autopilot/src/decoded_settlement.rs:115]: So AFAIU, for objective value computation we use solver fees `Trade::executed_solver_fee()` while the `executed_fee_amount` you mentioned is based on the regular fee `Trade::executed_fee()` which are different so I can't use that.

**Review by nlordell — APPROVED** [review]

File: crates/autopilot/src/decoded_settlement.rs (hunk 1 above)
Comment (nlordell) [reply crates/autopilot/src/decoded_settlement.rs:115]: Makes sense.

**Comment by sunce86** [comment]
> In theory, you could use the flag "is partially fillable" bit from the flags to check whether or not it is the trade's executed amount should be used or the full buy/sell amount.

Ah I get it. Ok, I will create additional issue for this and do it afterwards, since I want to merge the solver rewards PR asap.

**Pull request merged**